tests/lib/openstack_query/handlers/test_client_side_handler.py

```diff
@@ -0,0 +1,207 @@
+                MockProperties.PROP_1,
+                mock_prop_func,
+                mock_kwargs,
+            )
+
+        # when the preset is valid, but property is invalid
+        with self.assertRaises(QueryPresetMappingError):
+            self.instance.get_filter_func(
+                MockQueryPresets.ITEM_2,
+                MockProperties.PROP_3,
+                mock_prop_func,
+                mock_kwargs,
+            )
+
+        # when check_filter_func is false
+        mock_check_filter_func.return_value = False, "some-error"
+        with self.assertRaises(QueryPresetMappingError):
+            self.instance.get_filter_func(
+                MockQueryPresets.ITEM_1,
+                MockProperties.PROP_1,
+                mock_prop_func,
+                mock_kwargs,
+            )
+
+    def test_filter_func_wrapper(self):
+        mock_item = "some-openstack-item"
+        mock_filter_func = MagicMock()
+        mock_prop_func = MagicMock()
+        mock_filter_func_kwargs = {"arg1": "val1", "arg2": "val2"}
+        mock_filter_func.return_value = "a-boolean-value"
+
+        # when prop_func raises exception - i.e. property not found
+        mock_prop_func.side_effect = AttributeError()
+        res = self.instance._filter_func_wrapper(
+            mock_item, mock_filter_func, mock_prop_func, mock_filter_func_kwargs
+        )
+        self.assertFalse(res)
+
+        # when prop_func is valid and filter_kwargs given
+        mock_prop_func.return_value = "some-prop-val"
+        mock_prop_func.side_effect = None
+        res = self.instance._filter_func_wrapper(
+            mock_item, mock_filter_func, mock_prop_func, mock_filter_func_kwargs
+        )
+        self.assertEqual(res, "a-boolean-value")
+        mock_filter_func.assert_called_once_with(
+            "some-prop-val", **mock_filter_func_kwargs
+        )
+
+        # when prop_func is valid and filter_kwargs not given
+        mock_prop_func.reset_mock()
+        mock_filter_func.reset_mock()
+        mock_prop_func.return_value = "some-prop-val"
+        mock_prop_func.side_effect = None
+        res = self.instance._filter_func_wrapper(
+            mock_item, mock_filter_func, mock_prop_func
+        )
+        self.assertEqual(res, "a-boolean-value")
+        mock_filter_func.assert_called_once_with("some-prop-val")
+
+    @parameterized.expand(
+        [
+            ("required args only", {"arg1": 12}),
+            ("required and default", {"arg1": 12, "arg2": "non-default"}),
+            ("required and kwarg", {"arg1": 12, "some_kwarg": "some-val"}),
+            (
+                "all possible",
+                {"arg1": 12, "arg2": "non-default", "some_kwarg": "some-val"},
+            ),
+            (
+                "different order",
+                {"arg2": "non-default", "some_kwarg": "some-val", "arg1": 12},
+            ),
+        ]
+    )
+    def test_check_filter_func(self, name, valid_kwargs_to_test):
```

Comment:
I would argue this parametrised test is the pattern we should use to test invalid too, as we can add edge cases easily in the future too